demos/dotnet/dotnet/CscAction.cs

```diff
@@ -0,0 +1,103 @@
+// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System;
+using System.ComponentModel;
+using System.Diagnostics;
+using System.IO;
+using System.Text;
+
+namespace dotnet
+{
+    internal static class CscAction
+    {
+        public static bool Execute(ProjectProperties properties, Log log)
+        {
+            Console.WriteLine("compiling");
+            var processSettings = new ProcessStartInfo
+            {
+                FileName = properties.CscPath,
+                Arguments = properties.FormatCscArguments()
+            };
+
+            log.WriteLine("Executing {0}", processSettings.FileName);
+            log.WriteLine("Csc Arguments: {0}", processSettings.Arguments);
+
+            processSettings.CreateNoWindow = true;
+            processSettings.RedirectStandardOutput = true;
+            processSettings.UseShellExecute = false;
+
+            Process cscProcess;
+            try
+            {
+                cscProcess = Process.Start(processSettings);
+            }
+            catch (Win32Exception)
+            {
+                Console.WriteLine("ERROR: csc.exe needs to be on the path.");
+                return false;
+            }
+
+            if (cscProcess == null) return false;
+            var output = cscProcess.StandardOutput.ReadToEnd();
+            log.WriteLine(output);
+
+            cscProcess.WaitForExit();
+
+            return !output.Contains("error CS");
+        }
+
+        private static string FormatReferenceOption(this ProjectProperties project)
```

Comment:
This doesn't seem like the right place for these ProjectProperties formatters. ProjectPropertyCscExtensions or something...?